neo/Network/P2P/Payloads/Transaction.cs

```diff
@@ -18,16 +18,19 @@ namespace Neo.Network.P2P.Payloads
     public class Transaction : IEquatable<Transaction>, IInventory
     {
         public const int MaxTransactionSize = 102400;
+        public const uint MaxValidUntilBlockIncrement = 2102400;
         /// <summary>
         /// Maximum number of attributes that can be contained within a transaction
         /// </summary>
         private const int MaxTransactionAttributes = 16;
 
         public byte Version;
+        public uint Nonce;
         public byte[] Script;
         public UInt160 Sender;
         public long Gas;
         public long NetworkFee;
+        public uint ValidUntilBlock;
```

Comment:
No. This is a required field. After a few decades, our blockchain data will become extremely large, exceeding the size of the normal user's hard disk. At this time we have to trim some old transaction data. Once a transaction is trimmed, it may be used for replay attacks. So we either remember all the nones or let the old ones fail to pass the verification in the future. I chose to add the `ValidUntilBlock` field so that the old transactions could not be re-verified in the future, so that replay attacks can be avoided.